SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,841 @@
+    unsigned short inside = exit_code;
+    nodes->SetInsideTable(i_point, inside);
+    n_not_in_domain += exit_code;
+
+    /*--- Get lookup scalars ---*/
+    fluid_model_local->SetScalarLookups(scalars);
+    for (auto i_lookup = 0u; i_lookup<config->GetNLookups(); i_lookup++) {
+      nodes->SetLookupScalar(i_point, fluid_model_local->GetScalarLookups(i_lookup), i_lookup);
+    }
+
+    for (auto i_scalar = 0u; i_scalar < nVar; i_scalar++)
+      nodes->SetScalarSource(i_point, i_scalar, fluid_model_local->GetScalarSources(i_scalar));
+
+    su2double T = flowNodes->GetTemperature(i_point);
+    fluid_model_local->SetTDState_T(T,scalars);
+    /*--- set the diffusivity in the fluid model to the diffusivity obtained from the lookup table ---*/
+    for (auto i_scalar = 0u; i_scalar < nVar; ++i_scalar) {
+      nodes->SetDiffusivity(i_point, fluid_model_local->GetMassDiffusivity(i_scalar), i_scalar);
+    }
+
+    if (!Output) LinSysRes.SetBlock_Zero(i_point);
+  }
+
+  if (config->GetComm_Level() == COMM_FULL) {
+    SU2_MPI::Reduce(&n_not_in_domain, &global_table_misses, 1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE,
+                    SU2_MPI::GetComm());
+    if (rank == MASTER_NODE) {
+      SetNTableMisses(global_table_misses);
+    }
+  }
+
+  /*--- Clear residual and system matrix, not needed for
+   * reducer strategy as we write over the entire matrix. ---*/
+  if (!ReducerStrategy && !Output) {
+    LinSysRes.SetValZero();
+    if (implicit)
+      Jacobian.SetValZero();
+    else {
+      SU2_OMP_BARRIER
+    }
+  }
+
+  /*--- Upwind second order reconstruction and gradients ---*/
+
+  if (config->GetReconstructionGradientRequired()) {
+    if (config->GetKind_Gradient_Method_Recon() == GREEN_GAUSS) SetSolution_Gradient_GG(geometry, config, true);
+    if (config->GetKind_Gradient_Method_Recon() == LEAST_SQUARES) SetSolution_Gradient_LS(geometry, config, true);
+    if (config->GetKind_Gradient_Method_Recon() == WEIGHTED_LEAST_SQUARES)
+      SetSolution_Gradient_LS(geometry, config, true);
+  }
+
+  if (config->GetKind_Gradient_Method() == GREEN_GAUSS) SetSolution_Gradient_GG(geometry, config);
+
+  if (config->GetKind_Gradient_Method() == WEIGHTED_LEAST_SQUARES) SetSolution_Gradient_LS(geometry, config);
+
+}
+
+void CSpeciesFlameletSolver::Postprocessing(CGeometry* geometry, CSolver** solver_container, CConfig* config,
+                                            unsigned short iMesh) {
+  /*--- your postprocessing goes here ---*/
+}
+
+void CSpeciesFlameletSolver::SetInitialCondition(CGeometry** geometry, CSolver*** solver_container, CConfig* config,
+                                                 unsigned long ExtIter) {
+  const bool Restart = (config->GetRestart() || config->GetRestart_Flow());
+
+  if ((!Restart) && ExtIter == 0) {
+    if (rank == MASTER_NODE) {
+      cout << "Initializing progress variable and total enthalpy (using temperature)" << endl;
+    }
+
+    su2double* scalar_init = new su2double[nVar];
```

Comment:
use a vector type

Reply:
and when we pass the vector to SetSolution, we just use scalar_init.data() ?

Reply:
Yep